### PR TITLE
add automation coverage for BZ 2161993

### PR DESCRIPTION
**Pull request by vijaysawant**
### Problem Statement
Add automation coverage around the [Bug 2161993](https://bugzilla.redhat.com/show_bug.cgi?id=2161993) - wrong metadata if uploaded rpm have different name than name in rpm

### Solution
1. Create product, custom repo and upload rpm into repo
2. Create CV, add repo into it & publish CV
3. Register host with satellite and install rpm on it

### Related Issues
No

### CODE REVIEW

**Review by vsedmik — COMMENTED** [review]
Does it make sense to have `tests/foreman/data/ant-7.7.7-1.noarch.rpm` already renamed?

**Comment by vijaysawant** [comment]
> Does it make sense to have `tests/foreman/data/ant-7.7.7-1.noarch.rpm` already renamed?

`ant-7.7.7-1.noarch.rpm` this is the name of rpm file present at http://infra-podman-01.infra.sat.rdu2.redhat.com:50123/fake_yum3/

**Comment by vijaysawant** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_repository.py -k test_possitive_install_uploaded_rpm_on_host

**Comment by vsedmik** [comment]
> > Does it make sense to have `tests/foreman/data/ant-7.7.7-1.noarch.rpm` already renamed?
> 
> `ant-7.7.7-1.noarch.rpm` this is the name of rpm file present

I thought have it renamed within this PR to something else already, so that we don't rename it in the code. Instead of `tests/foreman/data/ant-7.7.7-1.noarch.rpm` you would be pushing `tests/foreman/data/noname-animal.noarch.rpm` or whatever you like.

**Review by ogajduse — CHANGES_REQUESTED** [review]

**Comment by ogajduse** [comment]
> > > Does it make sense to have `tests/foreman/data/ant-7.7.7-1.noarch.rpm` already renamed?
> > 
> > 
> > `ant-7.7.7-1.noarch.rpm` this is the name of rpm file present
> 
> I thought have it renamed within this PR to something else already, so that we don't rename it in the code. Instead of `tests/foreman/data/ant-7.7.7-1.noarch.rpm` you would be pushing `tests/foreman/data/noname-animal.noarch.rpm` or whatever you like.

I got your point, Vladimir. I like the way it is implemented now, the package (file name) still preserves the contextual info about what package it is. We could rename it to something like `our-renamed-ant-7.7.7-1.noarch.rpm`, however, I find an advantage in using the `genstring` function since it brings randomness to the name = more combinations tested across multiple test runs.

I do not have a strong opinion on this, however.

File: tests/foreman/cli/test_repository.py [hunk 1]
```diff
@@ -3390,3 +3394,87 @@ def test_positive_syncable_yum_format_repo_import(target_sat, module_org):
     )
     assert repodata['content-counts']['packages'] != 0
     assert repodata['sync']['status'] == 'Success'
+
+
+@pytest.mark.no_containers
+def test_possitive_install_uploaded_rpm_on_host(
+    target_sat, rhel7_contenthost, function_org, function_lce
```
Comment (ogajduse) [inline tests/foreman/cli/test_repository.py:3401]: Could you look around if we can fit the test logic in an existing test? This is completely non-blocking, especially if we save time using container hosts.

Comment (ogajduse) [reply tests/foreman/cli/test_repository.py:3401]: After a short research of our tests directory, I did not find any suitable test to put the logic in.

File: tests/foreman/cli/test_repository.py [hunk 2]
```diff
@@ -3390,3 +3394,87 @@ def test_positive_syncable_yum_format_repo_import(target_sat, module_org):
...
+    )
+    assert (
+        rhel7_contenthost.register(function_org, None, activation_key.name, target_sat).status == 0
+    )
+    assert (
+        rhel7_contenthost.execute(f'yum install -y {FAKE_3_YUM_REPO_RPMS[0].split("-")[0]}').status
+        == 0
+    )
+    installed_package_name = rhel7_contenthost.execute(
+        f'rpm -qa {FAKE_3_YUM_REPO_RPMS[0].split("-")[0]}'
```
[the remark this answers is not in the document]
Comment (vijaysawant) [reply tests/foreman/cli/test_repository.py:3478]: `rpm -qa` gives list of installed packages, but we can use `rpm -q` which will give information, thanks for correction.

**Comment by vijaysawant** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_repository.py -k test_possitive_install_uploaded_rpm_on_host

File: tests/foreman/cli/test_repository.py [hunk 3]
```diff
@@ -3390,3 +3394,87 @@ def test_positive_syncable_yum_format_repo_import(target_sat, module_org):
...
+    sync_status = target_sat.cli.Repository.info({'id': repo['id']})['sync']['status']
+    assert sync_status == 'Success', f'Failed to sync {repo["name"]} repo'
+
+    # Upload package
+    target_sat.put(
+        local_path=DataFile.FAKE_3_YUM_REPO_RPMS_ANT,
+        remote_path=f"/tmp/{FAKE_3_YUM_REPO_RPMS[0]}",
+    )
+    # Rename uploaded package name
+    FAKE_3_YUM_REPO_RPMS[0].split('.')
```
[the remark this answers is not in the document]
Comment (vijaysawant) [reply tests/foreman/cli/test_repository.py]: done, resolving conversation

File: tests/foreman/cli/test_repository.py [hunk 4]
```diff
@@ -3390,3 +3394,87 @@ def test_positive_syncable_yum_format_repo_import(target_sat, module_org):
...
+    target_sat, rhel7_contenthost, function_org, function_lce
+):
+    """Verify that uploaded rpm successfully install on content host
+
+    :id: 8701782e-2d1e-41b7-a9dc-07325bfeaf1c
+
+    :steps:
+        1. Create product, custom repo and upload rpm into repo
+        2. Create CV, add repo into it & publish CV
+        3. Register host with satellite and install rpm on it
```
[the remark this answers is not in the document]
Comment (vijaysawant) [reply tests/foreman/cli/test_repository.py]: done, resolving conversation

**Comment by vijaysawant** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_repository.py -k test_possitive_install_uploaded_rpm_on_host

**Review by vsedmik — APPROVED** [review]

File: tests/foreman/cli/test_repository.py [hunk 5]
```diff
@@ -3390,3 +3391,89 @@ def test_positive_syncable_yum_format_repo_import(target_sat, module_org):
     )
     assert repodata['content-counts']['packages'] != 0
     assert repodata['sync']['status'] == 'Success'
+
+
+@pytest.mark.rhel_ver_list([9])
+def test_possitive_install_uploaded_rpm_on_host(
```
Comment (adarshdubey-star) [inline tests/foreman/cli/test_repository.py:3397]:
```suggestion
def test_positive_install_uploaded_rpm_on_host(
```

Comment (vijaysawant) [reply tests/foreman/cli/test_repository.py:3397]: good catch, thanks you.

**Review by adarshdubey-star — APPROVED** [review]

**Comment by vijaysawant** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_repository.py -k test_positive_install_uploaded_rpm_on_host

**Pull request merged**